include/neighbor/Runtime.h

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2018-2019, Michael P. Howard.
+// This file is released under the Modified BSD License.
+
+// Maintainer: mphoward
+
+#ifndef NEIGHBOR_RUNTIME_H_
+#define NEIGHBOR_RUNTIME_H_
+
+#include <functional>
+#include <utility>
+
+#include <cuda_runtime.h>
+#ifdef __CUDACC__
+#define GPUCC
+#endif
+
+namespace neighbor
+{
+namespace gpu
+{
+
+/* errors */
+typedef cudaError_t error_t;
+enum error
+    {
+    success=cudaSuccess
+    };
```

Comment:
The error spoofing is a little clumsy, but I had trouble coming up with another solution. `cudaError_t` is a typedef of an enum, and so even though it can be typedef'd again, it's values still keep their prefixes (e.g., `cudaSuccess` or `hipSuccess`).  My solution was to make a new error enum, and then pilfer the values from the CUDA or HIP enums, as appropriate. Then, we still need to be able to do comparison between two enums, and so we have to coerce the values into integers. (Note: I still need to add inequality test here.)

I think this works OK though because error handling can then still be done by wrapping native API calls that take an `error_t`. The alternative was defining a new class that takes the native error in its constructor, but I think this is worse because you need to constantly remember to wrap it and you also still need custom comparators.